pkg/pluginmanager/manager.go

```diff
@@ -87,24 +87,27 @@ func discoverPlugins(pd []configtypes.PluginDiscovery) ([]discovery.Discovered,
 	return discoverSpecificPlugins(pd, nil)
 }
 
-// discoverSpecificPlugins returns the available plugin matching the criteria, if the criteria is not nil.
+// discoverSpecificPlugins returns all plugins that match the specified criteria from all PluginDiscovery sources,
+// along with an aggregated error (if any) that occurred while creating the plugin discovery source or fetching plugins.
 func discoverSpecificPlugins(pd []configtypes.PluginDiscovery, criteria *discovery.PluginDiscoveryCriteria) ([]discovery.Discovered, error) {
 	allPlugins := make([]discovery.Discovered, 0)
+	errorList := make([]error, 0)
 	for _, d := range pd {
 		discObject, err := discovery.CreateDiscoveryFromV1alpha1(d, criteria)
 		if err != nil {
-			return nil, errors.Wrapf(err, "unable to create discovery")
+			errorList = append(errorList, errors.Wrapf(err, "unable to create discovery source for '%v', error: '%v'", discObject.Name(), err.Error()))
```

Comment:
We are already wrapping the error through `errors.Wrapf` so we don't need to add it to the text of the message.
If you leave it like this it looks like:
```
tz context create --name tmc12 --endpoint unstable22.tmc-dev.cloud.vmware.com --staging
[i] API token env var is set

[ok] successfully created a TMC context
[i] Checking for required plugins...
[!] unable to automatically sync the plugins from target context. Please run 'tanzu plugin sync' command to sync plugins manually, error: 'unable to list plugins from discovery source 'default-tmc12', error: 'API error, status code: 404': API error, status code: 404'
```
Notice that `API error, status code: 404` is printed twice.
If you change it we get:
```
tz context create --name tmc13 --endpoint unstable22.tmc-dev.cloud.vmware.com --staging
[i] API token env var is set

[ok] successfully created a TMC context
[i] Checking for required plugins...
[!] unable to automatically sync the plugins from target context. Please run 'tanzu plugin sync' command to sync plugins manually, error: 'unable to list plugins from discovery source 'default-tmc13': API error, status code: 404'
```

Reply:
that is good suggestion. here is error message after removing error info in `errors.Wrapf` :
```
❯ t  context create --name tmc24 --endpoint unstable22.tmc-dev.cloud.vmware.com --staging
[i] API token env var is set

[ok] successfully created a TMC context
[i] Checking for required plugins...
[!] unable to automatically sync the plugins from target context. Please run 'tanzu plugin sync' command to sync plugins manually, error: 'unable to list plugins from discovery source 'default-tmc24': API error, status code: 404'
```